graphql-ballerina/query_depth_validator.bal

```diff
@@ -0,0 +1,103 @@
+// Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import graphql.parser;
+
+class QueryDepthValidator{
+    private parser:DocumentNode documentNode;
+    private int queryDepth = 1;
+    private int maxQueryDepth = 0;
+    private int queryDepthLimit;
+    private ErrorDetail[] errors;
+
+    public isolated function init(parser:DocumentNode documentNode, int queryDepthLimit){
+        self.documentNode = documentNode;
+        self.queryDepthLimit = queryDepthLimit;
+        self.errors = [];
+    }
+
+    public isolated function validate() returns ErrorDetail[]? {
+        self.visitDocument(self.documentNode);
+        if (self.errors.length() > 0) {
+            return self.errors;
+        }
+    }
+
+    public isolated function visitDocument(parser:DocumentNode documentNode) {
+        parser:OperationNode[] operations = documentNode.getOperations();
+        foreach parser:OperationNode operationNode in operations {
+            self.visitOperation(operationNode);
+            if (self.maxQueryDepth > self.queryDepthLimit) {
```

Comment:
How could this condition become true?
If we want to stop the other operations from validating when we find an operation that exceeds the `maxQueryDepth`, we can use a boolean class variable, to set the **"mode"** of the visitor.
For example, say a variable `isMaxDepthExceeded`, which should be set to `false` at the init function and then set to `true` as soon as we find a query that exceeds the maximum depth. 

(But I think we should process the next operations as well and create errors for each operation.